pkg/edgeview/src/basics.go

```diff
@@ -585,6 +586,22 @@ func listRecursiveFiles(path, pattern string) ([]string, error) {
 	return jfiles, nil
 }
 
+// get the cluster api-server IP and port from kubeconfig file
+func getKubeServerIPandPort(kubeConfigFile string) (string, error) {
+	content, err := os.ReadFile(kubeConfigFile)
+	if err != nil {
+		return "", err
+	}
+
+	regex := regexp.MustCompile(`\s+server: https://([^ ]+)`)
+	matches := regex.FindStringSubmatch(string(content))
+	if len(matches) != 2 {
+		return "", fmt.Errorf("failed to find server in kubeconfig")
+	}
+
+	return strings.TrimSpace(matches[1]), nil
+}
```

Comment:
```suggestion
func getKubeServerIPandPort(kubeYamlFile string) (string, error) {
        config, err := clientcmd.Load([]byte(kubeYamlFile))
        if err != nil {
                return "", fmt.Errorf("loading of config failed: %v", err)
        }

        var server string
        for k, v := range config.Clusters {
                if v != nil && v.Server != "" {
                        server = v.Server
                }
        }
        if server == "" {
                return "", fmt.Errorf("no server found")
        }

        u, err := url.Parse(server)
        if err != nil {
                return "", fmt.Errorf("could not parse server url: %+v", err)

        }

        return u.Host, nil
}
```

and import `k8s.io/client-go/tools/clientcmd`.

Reply:
well, I would not like to import tons of vendor files into edgeview side, just because I want to avoid a know endpoint string, which is "127.0.0.1:6443", thus as a compromise, I'm using the regexp to get the string from the config file. I can go back to static const definition also.

Reply:
> I would not like to import tons of vendor files into edgeview side

What makes it different to pillar, where we actually do this?

The `static const definition` would suffer from the lack of regression testing, too, doesn't it?

Reply:
the edgeview container currently has no use for kubernetes functions. This static const yes it may change later. but it does not change by itself. We are the one responsible for upgrade the k3s/kubernetes version, which is a big deal, it needs to be extensively tested, the top thing if this line changes, will break so many things, include this edgeview debugging tool. we hard code many things in const, many fall into this area (something might change later)

Reply:
i originally did use this kube client library, then i found it needs to import tons of things into edgeview/vendor side, then I changed this into using the search and match. I think it's good enough. 1) this line never changes in kubernetes, 2) if it changes later, other things are already been broken in the kubevirt functions 3) we are the one make the change, extensive testings are done with new version changes. I don't see problems here.

Reply:
I think the essence of this discussion is now that:
1. You say that adding the kubernetes dependencies are too many
2. I say it is okay as we add them to eve already

I don't have any more arguments to add, so if you want to parse it with a regex, then do.

Reply:
thanks! :-) also this 'edgeview' container is not only running on the EVE device, but it can be other places, the client running on laptop is the same 'edgeview' container.